db/migrate/20190116182055_add_discovery_tables.rb

```diff
@@ -112,50 +112,54 @@ def up
           CONSTRAINT commits_author_vcs_user_id_fkey FOREIGN KEY (author_vcs_user_id) REFERENCES vcs_users(id),
           CONSTRAINT commits_committer_vcs_user_id_fkey FOREIGN KEY (committer_vcs_user_id) REFERENCES vcs_users(id)
         );
-        
+
         CREATE
             TRIGGER trigger_set_timestamp BEFORE INSERT
                 OR UPDATE
                     ON
                     commits FOR EACH ROW EXECUTE PROCEDURE trigger_set_timestamp();
-        
+
         CREATE TABLE IF NOT EXISTS branches (
           id serial NOT NULL,
           branch_name text NOT NULL,
           branch_type branch_type NULL,
           repo_id int4 NULL,
+          project_id int4 NULL,
           created_at timestamp NOT NULL,
           updated_at timestamp NOT NULL,
           CONSTRAINT branches_pkey PRIMARY KEY (id),
-          CONSTRAINT branches_repo_id_fkey FOREIGN KEY (repo_id) REFERENCES repos(id)
+          CONSTRAINT branches_repo_id_fkey FOREIGN KEY (repo_id) REFERENCES repos(id) ON DELETE CASCADE,
```

Comment:
Does Delete cascade work both ways? For example, if we delete a branch, does DELETE CASCADE delete the repo associated to it? If so, I don't think we want that behavior.

Reply:
Only goes one way. So when a repo get's deleted, it will delete all the branches. Same thing for when a project get's deleted, the repo gets deleted, and then the branches, nodes and so on, cascading all the way down.

Reply:
👍 perfect.